#### Скачать данные из соревнования, пока можете работать только с train.csv. Считать их.

Посчитать основные статистики по каждому признаку (минимальные, максимальные, средние значения). Сделать выводы об адекватности этих статистик.

Посчитать для каждого признака, сколько в нем уникальных значений и как часто они встречаются. Сделать выводы по каждому признаку.

Используя выводы из двух предыдущих заданий, найдите те наблюдения, где есть пропуски, выбросы (неадекватные значения признака, которые отличаются от большинства), используя фильтрацию из pandas.

#### Data fields

**Id** - идентификационный номер квартиры

**DistrictId** - идентификационный номер района

**Rooms** - количество комнат

**Square** - площадь

**LifeSquare** - жилая площадь

**KitchenSquare** - площадь кухни

**Floor** - этаж

**HouseFloor** - количество этажей в доме

**HouseYear** - год постройки дома

**Ecology_1, Ecology_2, Ecology_3** - экологические показатели местности

**Social_1, Social_2, Social_3** - социальные показатели местности

**Healthcare_1, Helthcare_2** - показатели местности, связанные с охраной здоровья

**Shops_1, Shops_2** - показатели, связанные с наличием магазинов, торговых центров

**Price** - цена квартиры


In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('./train.csv')
df.head()

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
0  11809          27    3.0  115.027311         NaN           10.0      4   
1   3013          22    1.0   39.832524   23.169223            8.0      7   
2   8215           1    3.0   78.342215   47.671972           10.0      2   
3   2352           1    1.0   40.409907         NaN            1.0     10   
4  13866          94    2.0   64.285067   38.562517            9.0     16   

   HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  Social_2  \
0        10.0       2014   0.075424         B         B        11      3097   
1         8.0       1966   0.118537         B         B        30      6207   
2        17.0       1988   0.025609         B         B        33      5261   
3        22.0       1977   0.007122         B         B         1       264   
4        16.0       1972   0.282798         B         B        33      8667   

   Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2          Price  
0         0           NaN            0        0       B  305018.871089  
1         1        1183.0            1        0       B  177734.553407  
2         0         240.0            3        1       B  282078.720850  
3         0           NaN            0        1       B  168106.007630  
4         2           NaN            0        6       B  343995.102962

In [3]:
df[(df['Rooms'] == 0)]

Id  DistrictId  Rooms      Square  LifeSquare  KitchenSquare  Floor  \
2194   3159          88    0.0   38.697117   19.345131            9.0      9   
3780   7317          27    0.0   41.790881         NaN            0.0     13   
5894    770          28    0.0   49.483501         NaN            0.0     16   
6401   9443          27    0.0   87.762616   85.125471            0.0      5   
7644   7917          27    0.0  212.932361  211.231125            0.0      2   
7929    456           6    0.0   81.491446         NaN            0.0      4   
9767  12638          27    0.0  138.427694  136.215499            0.0      4   
9869   3224          27    0.0    2.377248    0.873147            0.0      1   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
2194        16.0       1982   0.127376         B         B        43   
3780         0.0       1977   0.211401         B         B         9   
5894         0.0       2015   0.118537         B         B        30   
6401        15.0       1977   0.211401         B         B         9   
7644         3.0       2008   0.211401         B         B         9   
7929         0.0       1977   0.243205         B         B         5   
9767         3.0       2016   0.075424         B         B        11   
9869         0.0       1977   0.017647         B         B         2   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
2194      8429         3           NaN            3        9       B   
3780      1892         0           NaN            0        1       B   
5894      6207         1        1183.0            1        0       B   
6401      1892         0           NaN            0        1       B   
7644      1892         0           NaN            0        1       B   
7929      1564         0         540.0            0        0       B   
9767      3097         0           NaN            0        0       B   
9869       469         0           NaN            0        0       B   

              Price  
2194  158998.110646  
3780   98129.976788  
5894  217009.338463  
6401  219281.918007  
7644  302211.260887  
7929  212864.799112  
9767  268394.744389  
9869  126596.941798

In [4]:
stat_tbl = df.describe().T
stat_tbl

count           mean            std           min  \
Id             10000.0    8383.407700    4859.019020      0.000000   
DistrictId     10000.0      50.400800      43.587592      0.000000   
Rooms          10000.0       1.890500       0.839512      0.000000   
Square         10000.0      56.315775      21.058732      1.136859   
LifeSquare      7887.0      37.199645      86.241209      0.370619   
KitchenSquare  10000.0       6.273300      28.560917      0.000000   
Floor          10000.0       8.526700       5.241148      1.000000   
HouseFloor     10000.0      12.609400       6.775974      0.000000   
HouseYear      10000.0    3990.166300  200500.261427   1910.000000   
Ecology_1      10000.0       0.118858       0.119025      0.000000   
Social_1       10000.0      24.687000      17.532614      0.000000   
Social_2       10000.0    5352.157400    4006.799803    168.000000   
Social_3       10000.0       8.039200      23.831875      0.000000   
Healthcare_1    5202.0    1142.904460    1021.517264      0.000000   
Helthcare_2    10000.0       1.319500       1.493601      0.000000   
Shops_1        10000.0       4.231300       4.806341      0.000000   
Price          10000.0  214138.857399   92872.293865  59174.778028   

                         25%            50%            75%           max  
Id               4169.500000    8394.500000   12592.500000  1.679800e+04  
DistrictId         20.000000      36.000000      75.000000  2.090000e+02  
Rooms               1.000000       2.000000       2.000000  1.900000e+01  
Square             41.774881      52.513310      65.900625  6.410652e+02  
LifeSquare         22.769832      32.781260      45.128803  7.480592e+03  
KitchenSquare       1.000000       6.000000       9.000000  2.014000e+03  
Floor               4.000000       7.000000      12.000000  4.200000e+01  
HouseFloor          9.000000      13.000000      17.000000  1.170000e+02  
HouseYear        1974.000000    1977.000000    2001.000000  2.005201e+07  
Ecology_1           0.017647       0.075424       0.195781  5.218671e-01  
Social_1            6.000000      25.000000      36.000000  7.400000e+01  
Social_2         1564.000000    5285.000000    7227.000000  1.908300e+04  
Social_3            0.000000       2.000000       5.000000  1.410000e+02  
Healthcare_1      350.000000     900.000000    1548.000000  4.849000e+03  
Helthcare_2         0.000000       1.000000       2.000000  6.000000e+00  
Shops_1             1.000000       3.000000       6.000000  2.300000e+01  
Price          153872.633942  192269.644879  249135.462171  6.332335e+05

In [5]:
df.max()

Id                      16798
DistrictId                209
Rooms                    19.0
Square             641.065193
LifeSquare        7480.592129
KitchenSquare          2014.0
Floor                      42
HouseFloor              117.0
HouseYear            20052011
Ecology_1            0.521867
Ecology_2                   B
Ecology_3                   B
Social_1                   74
Social_2                19083
Social_3                  141
Healthcare_1           4849.0
Helthcare_2                 6
Shops_1                    23
Shops_2                     B
Price            633233.46657
dtype: object

При беглом просмотре датафрейма с помощью функции describe() можно сделать следующие предварительные выводы:
1. В признаках LifeSquare и Healthcare_1 имеются пропуски наблюдений, т.к. количество наблюдений по этим признакам не совпадает с полным количесвтом наблюдений.
2. Статистические значения mean, std, min, max, медиану и квартили можно рассматривать только по следующим признакам: Rooms, Square, LifeSquare, KitchenSquare, Floor, Price. Что нам это дает? Ответ: среднюю температуру по больнице. Что бы оперировать этими значениями значениями, нужно выполнить дополнительные группировки по признакам. Только тогда, возможно, это будет иметь смысл.
3. Имеются выбросы по следующим признакам:
    - Rooms - количесвто комнат = 0;
    - Square - многое слышал про Москву и стоимость жилья там, но смоневаюсь, что там имеются квартиры общей площадью 1.14 м2;
    - LifeSquare - аналогично предыдущему пункту: жилая площадь 0.37м2 не укладывается ни в какие нормы;
    - KitchenSquare - кухня она, либо есть, либо ее нет, но не S = 0.0м2;
    - HouseFloor - не бывает домов с этажностью равной 0, к тому же минимальный этаж среди всех продаваемых квартир Floor равен 1;
    - HouseFloor - количество этажей = 117. Специально загуглил: самое высокое здание в Москве имеет 95 этажей;
    - LifeSquare - максимальная жилая площадь 7480 значительно превышает общую площадь LifeSquare = 641;
    - KitchenSquare - аналогично предыдущему пункту;
    
    